packages/contracts-bedrock/src/universal/UsdcBridge.sol

```diff
@@ -0,0 +1,362 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeCall } from "src/libraries/SafeCall.sol";
+import { CrossDomainMessenger } from "src/universal/CrossDomainMessenger.sol";
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import { Pausable } from "src/libraries/Pausable.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IPartialUsdc } from "src/L1/IPartialUsdc.sol";
+
+/// @custom:upgradeable
+/// @title UsdcBridge
+/// @notice UsdcBridge is a base contract for the L1 and L2 standard ERC20 bridges. It handles
+///         the core bridging logic, including escrowing tokens that are native to the local chain
+///         and minting/burning tokens that are native to the remote chain.
+abstract contract UsdcBridge is Initializable, Pausable, Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @notice The L2 gas limit set when eth is depoisited using the receive() function.
+    uint32 internal constant RECEIVE_DEFAULT_GAS_LIMIT = 200_000;
+
+    /// @custom:legacy
+    /// @custom:spacer messenger
+    /// @notice Spacer for backwards compatibility.
+    bytes30 private spacer_0_2_30;
+
+    /// @custom:legacy
+    /// @custom:spacer l2TokenBridge
+    /// @notice Spacer for backwards compatibility.
+    address private spacer_1_0_20;
+
+    /// @notice Mapping that stores deposits for a given pair of local and remote tokens.
+    mapping(address => mapping(address => uint256)) public deposits;
+
+    /// @notice Messenger contract on this domain.
+    /// @custom:network-specific
+    CrossDomainMessenger public messenger;
+
+    /// @notice Corresponding bridge on the other domain.
+    /// @custom:network-specific
+    UsdcBridge public otherBridge;
+
+    /// @notice Address of the token on L1.
+    address public l1Usdc;
+
+    /// @notice Address of the token on L2.
+    address public l2Usdc;
+
+    /// @notice Reserve extra slots (to a total of 50) in the storage layout for future upgrades.
+    ///         A gap size of 43 was chosen here, so that the first slot used in a child contract
+    ///         would be a multiple of 50.
+    uint256[43] private __gap;
```

Comment:
Why did you change this?

Reply:
There are 2 new storage slots being used before this. By reducing the gap, the multiple is kept to 50 as it was saying in the original comment. I don't think it matters much though

Reply:
The was actually a necessary change. If we left it at 45 and later add 45 more slots we might (instead of 43), then we would have overridden the existing storage for other contracts that are serialized after the USDC bridge in the inheritance hierarchy.

Reply:
The was actually a necessary change. If we left it at 45, then we would have caused a storage shift for other contracts that are serialized after the USDC bridge in the inheritance hierarchy (I haven't checked if there were other contracts after the USDC bridge, but changing it to 43 was the safe choice. ) 

From:
https://docs.openzeppelin.com/contracts/3.x/upgradeable#:~:text=You%20may%20notice%20that%20every,storage%20compatibility%20with%20existing%20deployments.

> It isn’t safe to simply add a state variable because it "shifts down" all of the state variables below in the inheritance chain.